server/src/services/formation.service.ts

```diff
@@ -0,0 +1,870 @@
+import { FormationCatalogue } from "../common/model/index.js"
+import { IApiError, manageApiError } from "../common/utils/errorManager.js"
+import { regionCodeToDepartmentList } from "../common/utils/regionInseeCodes.js"
+import { trackApiCall } from "../common/utils/sendTrackingEvent.js"
+import { sentryCaptureException } from "../common/utils/sentryUtils.js"
+import { notifyToSlack } from "../common/utils/slackUtils.js"
+import config from "../config.js"
+import { formationMock, formationsMock, formationDetailMock } from "../mocks/formations-mock.js"
+import { formationsQueryValidator, formationsRegionQueryValidator } from "../service/formationsQueryValidator.js"
+import { IFormationEsResult } from "./formation.service.types.js"
+import { ILbaItem, ILbaItemTrainingSession, LbaItem } from "./shared.service.types.js"
+
+const formationResultLimit = 500
+
+const lbfDescriptionUrl = "https://labonneformation.pole-emploi.fr/api/v1/detail"
+
+const esClient = getElasticInstance()
+
+const diplomaMap = {
+  3: "3 (CAP...)",
+  4: "4 (BAC...)",
+  5: "5 (BTS, DEUST...)",
+  6: "6 (Licence, BUT...)",
+  7: "7 (Master, titre ingénieur...)",
+}
+
+const getDiplomaIndexName = (value) => {
+  if (value) {
+    return diplomaMap[value[0]]
+  } else {
+    return ""
+  }
+}
+
+/**
+ * Récupère les formations matchant les critères en paramètre depuis Elasticsearch
+ * @param {string[]} romes un tableau de codes ROME
+ * @param {string} romeDomain un domaine de ROME
+ * @param {[number, number]} coords des coordonnées au format [longitude,latitude] pour centrer la recherche
+ * @param {number} radius le rayon de recherche
+ * @param {string} diploma le niveau de diplôme visé
+ * @param {number} limit le nombre de résultats max à produire
+ * @param {string} caller l'identifiant fourni par l'exploitant de l'api
+ * @param {string} api le nom de l'api liée à cette fonction
+ * @param {string[]} options un tableau d'options modulant la recherche
+ * @param {string} useMock un flag indiquant s'il faut retourner des données mockées ou non
+ * @returns {Promise<IFormationEsResult[]>}
+ */
+export const getFormations = async ({
+  romes,
+  romeDomain,
+  coords,
+  radius,
+  diploma,
+  limit,
+  caller,
+  api = "formationV1",
+  options,
+  useMock,
+}: {
+  romes?: string[]
+  romeDomain?: string
+  coords?: [string, string]
+  radius?: number
+  diploma?: string
+  limit?: number
+  caller?: string
+  api?: string
+  options: string[]
+  useMock: string
+}): Promise<IFormationEsResult[]> => {
+  try {
+    if (useMock && useMock !== "false") {
+      return formationsMock
+    }
+
+    const distance = radius || 30
+
+    const useGeoLocation = coords ? true : false
+    const latitude = coords ? coords[1] : null
+    const longitude = coords ? coords[0] : null
+
+    const now: Date = new Date()
+    const tags: number[] = [now.getFullYear(), now.getFullYear() + 1, now.getFullYear() + (now.getMonth() < 8 ? -1 : 2)]
```

Comment:
```suggestion
    const now = new Date()
    const tags = [now.getFullYear(), now.getFullYear() + 1, now.getFullYear() + (now.getMonth() < 8 ? -1 : 2)]
```

Préférer l'inférence TS pour les variables si ce sont pas des objets/arrays complexes pour lesquels le type n'a pas été passé par l'origine